wled00/playlist.cpp

```diff
@@ -10,19 +10,73 @@ typedef struct PlaylistEntry {
   uint16_t tr;
 } ple;
 
-byte playlistRepeat = 1;
-byte playlistEndPreset = 0;
+int8_t   playlistRepeat = 1;
+byte     playlistEndPreset = 0;
+byte    *playlistEntries = nullptr;
+byte     playlistLen;
+int8_t   playlistIndex = -1;
+uint16_t playlistEntryDur = 0;
 
-uint8_t* playlistEntries;
 
-byte playlistLen;
-int8_t playlistIndex = -1;
+void shufflePlaylist() {
+  int currentIndex = playlistLen, randomIndex;
+
+  PlaylistEntry temporaryValue, *entries = reinterpret_cast<PlaylistEntry*>(playlistEntries);
+
+  // While there remain elements to shuffle...
+  while (currentIndex--) {
+    // Pick a random element...
+    randomIndex = random(0, currentIndex);
+    // And swap it with the current element.
+    temporaryValue = entries[currentIndex];
+    entries[currentIndex] = entries[randomIndex];
+    entries[randomIndex] = temporaryValue;
+  }
+}
+
+/*
+ * The same thing as saving and loading playlist can be achieved using JSON API saved in a preset.
+ * 
+void deserializePlaylist() {
+  DynamicJsonDocument doc(JSON_BUFFER_SIZE);
+
+  DEBUG_PRINTLN(F("Reading playlist from /playlist.json..."));
+
+  if (!readObjectFromFile("/playlist.json", nullptr, &doc)) return; //if file does not exist just exit
+
+  JsonObject playlist = doc[F("playlist")];
+  if (!playlist.isNull()) loadPlaylist(playlist);
+}
```

Comment:
is this lengthy comment needed? if so: why do we want it to be here?

Reply:
It is a code remnant if you'd want to have loading and saving of playlists to FS. It can be deleted.